sql2.txt

```diff
@@ -17,29 +27,45 @@ may have ordered the same product in several different orders.  So the total qua
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
 
-
+SELECT CustomerName, ProductName, SUM(od.Quantity) AS TotalQuantity 
+FROM Products AS p 
+JOIN orderDetails AS od ON p.ProductID=od.ProductID 
+JOIN orders AS o  ON o.orderID=od.orderID 
+JOIN Customers AS c on c.CustomerID = o.CustomerID 
```

Comment:
It's interesting that the order of your joins has been reversed compared to the previous answer, as the instructions say to "Extend the previous query", but it's not that important because the result set is the same.

```suggestion
FROM Customers AS c 
JOIN  orders AS o  ON c.CustomerID=o.CustomerID 
JOIN orderDetails AS od ON o.orderID=od.orderID
JOIN Products AS p ON p.ProductID=od.ProductID
```

Reply:
Done) 
CustomerName  confused me)